src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java

```diff
@@ -166,4 +177,78 @@ public void testFinishLocalSetup() {
     }
 
+    /**
+     * Verifies that the language specified with the system property {@code user.language} exists.
+     */
+    @Test
+    public void testLanguageIsValid() {
+        final String language = DEFAULT_LOCALE.getLanguage();
+        if (!language.isEmpty()) {
+            assertThat("Invalid language",
+                    Arrays.asList(Locale.getISOLanguages()), hasItem(language));
+        }
+    }
+
+    /**
+     * Verifies that the country specified with the system property {@code user.country} exists.
+     */
+    @Test
+    public void testCountryIsValid() {
+        final String country = DEFAULT_LOCALE.getCountry();
+        if (!country.isEmpty()) {
+            assertThat("Invalid country",
+                    Arrays.asList(Locale.getISOCountries()), hasItem(country));
+        }
+    }
+
+    /**
+     * Verifies that if the language is specified explicitly (and it is not English),
+     * the message does not use the default value.
+     */
+    @Test
+    public void testLocaleIsSupported() {
+        final String language = DEFAULT_LOCALE.getLanguage();
+        if (!language.isEmpty() && !Locale.ENGLISH.getLanguage().equals(language)) {
+            final DefaultLogger.LocalizedMessage message = createSampleLocalizedMessage();
+            assertThat("Unsupported language: " + DEFAULT_LOCALE,
+                    message.getMessage(), not("Error auditing {0}"));
+        }
+    }
+
+    @DefaultLocale("fr")
+    @Test
+    public void testCleatBundleCache() {
+        final DefaultLogger.LocalizedMessage message = createSampleLocalizedMessage();
+
+        assertEquals("Une erreur est survenue {0}", message.getMessage(), "Invalid message");
+
+        final Map<String, ResourceBundle> bundleCache =
+                Whitebox.getInternalState(DefaultLogger.LocalizedMessage.class, "BUNDLE_CACHE");
+
+        assertEquals(1, bundleCache.size(), "Invalid bundle cache size");
+    }
+
+    @Test
+    public void testNullArgs() {
+        final DefaultLogger.LocalizedMessage message = new DefaultLogger.LocalizedMessage(
+                DefaultLogger.ADD_EXCEPTION_MESSAGE, new String[] {"myfile"});
+        final String output = message.getMessage();
+        assertTrue(output.contains("Error auditing myfile"),
+                "Violation should contain exception info, but was " + output);
+
+        final DefaultLogger.LocalizedMessage messageWithNullArgs =
+                new DefaultLogger.LocalizedMessage(DefaultLogger.ADD_EXCEPTION_MESSAGE);
+        final String outputForNullArgs = messageWithNullArgs.getMessage();
+        assertTrue(outputForNullArgs.contains("Error auditing {0}"),
+                "Violation should contain exception info, but was " + outputForNullArgs);
+    }
+
+    private static DefaultLogger.LocalizedMessage createSampleLocalizedMessage() {
+        return new DefaultLogger.LocalizedMessage(DefaultLogger.ADD_EXCEPTION_MESSAGE);
+    }
+
+    @AfterEach
+    public void tearDown() {
```

Comment:
move up to be before first method with `@Test`

Reply:
This is removed since `clearCache()` method is removed as it was not being used in the `DefaultLogger` class.

Reply:
It was added again. I have some doubt regarding this method. This method is not being called anywhere. When I remove this, test fails. First is https://github.com/checkstyle/checkstyle/blob/567d85eddcd7336334f41543602481304b580f23/src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java#L260 fails. 

And second, the major one is pitest fails for `getResourceBundle`, [pitest-failure-report](https://91454-12499251-gh.circle-artifacts.com/2/target/pit-reports/202106212158/com.puppycrawl.tools.checkstyle/DefaultLogger.java.html#grouporg.pitest.mutationtest.report.html.SourceFile@604447f8_300)